### ISSUE TITLE
crash on exit after change of replication source

### ISSUE THREAD

**Issue by kostja** [tarantool/tarantool#574]
The following crashes in coeio_custom:
```
ostja@ibbur ~/work/tarantool/test/app
 % git diff
diff --git a/test/app/cfg.test.lua b/test/app/cfg.test.lua
index 9271620..c0999b4 100755
--- a/test/app/cfg.test.lua
+++ b/test/app/cfg.test.lua
@@ -61,5 +61,12 @@ test:is(box.cfg.wal_mode, "none", "cfg.wal_mode change")
 box.cfg{wal_mode = require('msgpack').NULL}
 test:is(box.cfg.wal_mode, "write", "cfg.wal_mode default value")
 
+--------------------------------------------------------------------------------
+-- gh-537: segmentation fault with replication_source
+--------------------------------------------------------------------------------
+
+box.cfg{replication_source = 'guest:password@localhost:3301'}
+box.cfg{replication_source = ""}
+
 test:check()
 os.exit(0)
```
My guess is that attempting to cancel a fiber which is inside coeio_custom task resumes
that fiber, and destroys it, while  eio thread is reading its stack.
The fix is to delay cancellation until a coeio task has completed, but for some reason it didn't work.

**Comment by rtsisyk** [comment]
I think we should fix coeio_custom #364 first and when return to this ticket.

**Comment by rtsisyk** [comment]
Re-test after merging multi-master #729
